source/Coop.Core/Client/States/LoadingState.cs

```diff
@@ -43,6 +46,7 @@ public override void Connect()
 
         public override void Disconnect()
         {
+            MessageBroker.Publish(this, new EnterMainMenu());
```

Comment:
Every state should subscribe to MainMenuEntered and transition to the main menu state when that message is received.